master/internal/restore_test.go

```diff
@@ -0,0 +1,17 @@
+package internal
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestShimExperimentSnapshotV4(t *testing.T) {
```

Comment:
This looks good.

Now that we have a test with an embedded snapshot, can we write a test with a v5 snapshot, and ensure that it is loadable into the current system?  I feel like this sort of bug ought to be preventable.

Reply:
Yeah, good idea, let me take a shot at that.